### DISCUSSION TITLE
Workflow/actions status badge giving 404 on private repo in an organization

### DISCUSSION THREAD

**Discussion by Siedlerchr** [community/community#26812] [Actions]
<p>I tried to put a GitHub Actions workflow badge ony a private repo, however, it’s always giving me 404</p>
<pre><code>![](https://github.com/&lt;org&gt;/&lt;repoName&gt;/.github/workflows/main.yml/badge.svg)
</code></pre>
<p>Don’t badges work on private repos?<br>
The workflow’s file is correct.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hey <a class="mention" href="/siedlerchr">@siedlerchr</a>,</p>
<p>Thanks for reaching out and for taking the time to submit this report, our engineering team is working on it and I’ll post an update here once I have some news.</p>

**Comment by ghost** [comment]
<p>There is no any information about badges works on private repos or not on <a href="https://help.github.com/en/articles/configuring-a-workflow#adding-a-workflow-status-badge-to-your-repository" rel="nofollow noopener">Docs</a>. And currently it returns 404</p>

**Comment by mscoutermarsh** [comment, marked as answer]
<p>Try replacing: </p>
<pre><code>.github/workflows/main.yml
</code></pre>
<p>With the name of your workflow.</p>
<p>Path is only used on workflows that don’t have a name.</p>
<p>Sorry that’s confusing. We’ll be adding a button to autogenerate these URLs for you in the near future.</p>

**Comment by Siedlerchr** [comment]
<p>I tried it today and it works.<br>
Except for linking, it would be nice if a click on the badge would lead to the actions tab.</p>

**Comment by smikitky** [comment]
<p>Recently I ran into a similar problem (tried to use the path syntax for a <strong>named</strong> workflow).</p>
<p>But is there a specific reason why we can’t we use the path syntax for a named workflow? Some workflows have names with non-English-based, URL-unfriendly characters, and there are cases where path-based URLs look cleaner.</p>

**Comment by stefan-kolb** [comment]
<p>I would request to be able to use both name or file path. The name of the workflow might change due to stylistic reasons but the path will most likely (or might) stay the same.</p>

**Comment by amalv** [comment]
<p>This solution didn’t work for me, I have a private repository that uses the named workflow:</p>
<pre><code>[![Actions Status](https://github.com/&lt;MY_USERNAME&gt;/&lt;REPOSITORY_NAME&gt;/workflows/Semantic%20release/badge.svg)](https://github.com/&lt;MY_USERNAME&gt;/&lt;REPOSITORY_NAME&gt;/actions)
</code></pre>
<p>The badge is being displayed, but it always shows “no status”.</p>
<p><img src="https://user-images.githubusercontent.com/1252707/181099996-f35f10f8-d6fa-430d-9960-ebc3230f5b3b.png" alt="noStatus.png" data-base62-sha1="hixMDNoRubbz1Y2tJRt4mjFui4C" width="202" height="30"></p>

**Comment by Siedlerchr** [comment]
<p>It takes some time until the badge is generated with the correct value.</p>
<p>For me it works now in a private repo</p>

**Comment by giankotarola** [comment]
<p><a class="mention" href="/siedlerchr">@siedlerchr</a> works for you by replacing</p>
<pre><code>.github/workflows/main.yml
</code></pre>
<p>with the name of the workflow as <a class="mention" href="/mscoutermarsh">@mscoutermarsh</a> suggested? </p>
<p>Like:</p>
<p><img src="https://github.com/%3CORG%3E/%3CREPO_NAME%3E/%3CWORKFLOW_NAME%3E/badge.svg" alt="Action Status"></p>

**Comment by awinecki** [comment]
<p>Doesn’t seem to work for me (displaying badges for a private repo).</p>
<pre><code class="lang-auto">https://github.com/&lt;org&gt;/&lt;repo&gt;/workflows/&lt;workflow-name&gt;/badge.svg
</code></pre>
<p>Works as long as I open it in a browser that has Github session open. If I open up the above link in the incognito mode, it returns 404 error.</p>
<p>I couldn’t find any way to add some authentication / token to that request. Similar issue on SO: <a href="https://stackoverflow.com/questions/62190387/getting-the-status-of-a-badge-from-a-private-repository" class="inline-onebox" rel="noopener nofollow ugc">github - Getting the status of a badge from a private repository - Stack Overflow</a>.</p>
<p>What I want to achieve is to have an internal developer portal app that displays CI status of each app. I could probably integrate it using API, but badges seem to be simple and do what I want. Perhaps they were not designed for such use cases.</p>

**Comment by KeithClarke** [comment]
<p>Hi <a class="mention" href="/mscoutermarsh">@mscoutermarsh</a>,</p>
<p>Not working for me either using the path or name for a private repo. The name URL works where there is a GitHub session, but not in an “incognito window”.</p>
<p>Also asked here <a href="https://stackoverflow.com/questions/67794007/does-github-support-anonymous-access-to-workflow-status-badges-for-private-repos" class="inline-onebox" rel="noopener nofollow ugc">continuous integration - Does Github support anonymous access to workflow status badges for private repositories? - Stack Overflow</a>.</p>

**Comment by andre-lx** [comment]
<p>Hi!</p>
<p>Working well in browser, but, if I use the badge_url from the API inside my html github dashboard, I always get the 404 (using the same browser where github is logged in). It doesn’t make sense.</p>
<p>In the HTML I got the 404 icon, but when I click the 404 icon and click “Open image in a new tab” it works as expected.</p>
<p>And this only happens for private repos.</p>
<p>Can you give some advice? Thanks.</p>

**Comment by edm-ms** [comment]
<p>I too am seeing the same behavior. No matter what I try, no matter what auth headers I use, I cannot download the badge.svg from a private repo. This is an issue, especially for folks who don’t live inside GitHub. We want to take these status badges to display in another dashboard, but we cannot get it working.</p>

**Comment by TheJakey** [comment]
Year later and issue persists... Can we reopen it or something?

**Reply by starkindustries** [reply]
Go to your repository. Click **Actions** then select your workflow. Click the three dots towards the top right next to the filter search text field. Then click create status badge. Copy/paste that code into your README and done.
https://dev.to/azure/github-how-to-display-the-status-badge-for-a-github-action-5449

**Reply by TheJakey** [reply]
Tried that, but this is what I get trying it on my Linux Build workflow 
It's a private repository which seems to be the issue. (has to be private - it's my deploma-thesis repo and that cannot be public until it's submitted and evaluated for plagiarism) - nonetheless it should be able to display that status imo. 

![image](https://user-images.githubusercontent.com/32109375/206900437-1bede5f1-6444-4dc8-9533-9ba9f3f919a4.png)

**Reply by TheJakey** [reply]
Okay I do feel little stupid rn .

Ofc it works

It doesn't show result in IDE, 'casue it's private and auth is apperently not happening there, but when commited and viewed via GitHub, it works. Thanks!

![image](https://user-images.githubusercontent.com/32109375/206900691-5ae08d7b-6c3a-47f3-8915-0b2ae0e20781.png)

_(P.S. yep, I need to fix some issues to pass the build, but step by step! :D )_

**Comment by mwotton** [comment]
so this is a tremendously cheesy answer, but you can script it with gh and shields.io. could potentially do it at the end of a workflow, but then it'd need to be customised for where it ends up.

```
#!/bin/bash

if [ "$#" -ne 1 ]; then
  echo "Usage: $0 repo_name" >&2
  exit 1
fi

REPO="$1"

# Fetch workflow list
workflow_list=$(gh api -X GET "repos/$REPO/actions/workflows" --jq '.')

if [ "$?" -ne 0 ]; then
  echo "Error fetching workflow list" >&2
  exit 1
fi

# Check if a "build" workflow exists
build_workflow=$(echo "$workflow_list" | jq '.workflows[] | select(.name == "build")')

if [ -n "$build_workflow" ]; then
  workflow_id=$(echo "$build_workflow" | jq '.id')
else
  # Use the first workflow
  first_workflow=$(echo "$workflow_list" | jq '.workflows[0]')

  if [ -z "$first_workflow" ]; then
    echo "No workflows found" >&2
    exit 1
  fi

  workflow_id=$(echo "$first_workflow" | jq '.id')
fi

# Fetch the latest workflow run
latest_workflow_run=$(gh api -X GET "repos/$REPO/actions/workflows/$workflow_id/runs?per_page=1" --jq '.')

if [ "$?" -ne 0 ]; then
  echo "Error fetching latest workflow run" >&2
  exit 1
fi

# Extract the latest workflow run conclusion and URL
latest_workflow_run_conclusion=$(echo "$latest_workflow_run" | jq -r '.workflow_runs[0].conclusion')
latest_workflow_run_url=$(echo "$latest_workflow_run" | jq -r '.workflow_runs[0].html_url')

# Set the badge color based on the conclusion
case "$latest_workflow_run_conclusion" in
  "success") badge_color="green" ;;
  "failure") badge_color="red" ;;
  *) badge_color="blue" ;;
esac

# Generate the badge URL using shields.io
badge_url="https://img.shields.io/badge/${REPO//_/-}-%20${latest_workflow_run_conclusion}-${badge_color}"

# Download the badge image and base64 encode it
base64_encoded_svg=$(curl -s -L "$badge_url" | base64 -w 0)

# Generate the Obsidian image and link
echo "[![$REPO](data:image/svg+xml;base64,$base64_encoded_svg)]($latest_workflow_run_url)"
```

**Comment by djcampbell** [comment]
I'm facing the same 404 issue referencing the badge from a private repo on Github Pages with a custom-generated URL. Problem is the Session Cookie Headers are being ignored/revoked because the source URL is not 'github.com'.

'This attempt to set a cookie via a Set-Cookie header was blocked because it had the "SameSite=Lax" attribute but came from a cross-site response which was not the response to a top-level navigation.'
